jena-text/src/main/java/org/apache/jena/query/text/TextQueryPF.java

```diff
@@ -210,44 +212,36 @@ private QueryIterator variableSubject(Binding binding, Node s, Node score, Node
             return bmap;
         } ;
         
-        Iterator<Binding> bIter = Iter.map(r.iterator(), converter);
+        Iterator<Binding> bIter = Iter.map(results.iterator(), converter);
         QueryIterator qIter = new QueryIterPlainWrapper(bIter, execCxt);
         return qIter ;
     }
 
+    private QueryIterator variableSubject(Binding binding, Node s, Node score, Node literal, StrMatch match, ExecutionContext execCxt) {
+        List<TextHit> r = query(match.getProperty(), match.getQueryString(), match.getLimit(), execCxt) ;
+        return resultsToQueryIterator(binding, s, score, literal, r, execCxt);
+    }
+
     private QueryIterator concreteSubject(Binding binding, Node s, Node score, Node literal, StrMatch match, ExecutionContext execCxt) {
         if (!s.isURI()) {
             log.warn("Subject not a URI: " + s) ;
             return IterLib.noResults(execCxt) ;
         }
 
-        Var scoreVar = (score==null) ? null : Var.alloc(score) ;
-        Var literalVar = (literal==null) ? null : Var.alloc(literal) ;
         String qs = match.getQueryString() ;
         List<TextHit> x = query(match.getProperty(), match.getQueryString(), -1, execCxt) ;
         
         if ( x == null ) // null return value - empty result
             return IterLib.noResults(execCxt) ;
         
+        List<TextHit> r = new LinkedList();
```

Comment:
Even if you plan to replace this, it would be good to get the fix in so the functionality is correct in master.  It gives work on the cache it's own space to happen.